prov/sm2/src/sm2_coordination.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Amazon.com, Inc. or its affiliates. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef _SM2_COORDINATION_H_
+#define _SM2_COORDINATION_H_
+
+#include "config.h"
+
+#include <stddef.h>
+#include <stdint.h>
+#include <sys/un.h>
+
+#include <ofi_atom.h>
+#include <ofi_hmem.h>
+#include <ofi_mem.h>
+#include <ofi_proto.h>
+#include <ofi_rbuf.h>
+#include <ofi_tree.h>
+
+#include <rdma/providers/fi_prov.h>
+
+#define SM2_NAME_MAX	      256
+#define SM2_INJECT_SIZE	      4096
+#define SM2_MAX_UNIVERSE_SIZE 4096
+/* TODO: Make the number of XFER ENTRY's configurable */
+#define SM2_NUM_XFER_ENTRY_PER_PEER 2048
+
+struct sm2_mmap {
+	char *base;
+	size_t size;
+	int fd;
+};
+
+struct sm2_ep_allocation_entry {
+	int pid; /* This is for allocation startup */
+	char ep_name[SM2_NAME_MAX];
+	uint32_t startup_ready; /* TODO Do I need to make atomic */
+};
+
+struct sm2_coord_file_header {
+	int file_version;
+	pthread_mutex_t write_lock;
+	ofi_atomic32_t pid_lock_hint;
```

Comment:
@lrbison correct me if I am wrong... but I believe this is atomic b/c we check it before we get the lock to see if the PID is dead.... I don't think we are using this feature heavily right now, so I think this can potentially go away?

Reply:
Removed PID hint and everything seems to be working (plus I think it fixed CI... TBD).

Reply:
Agree it can be non-atomic.